### ISSUE TITLE
Missing dependency after transpiling?

### ISSUE THREAD

**Issue by mrbannon** [paperjs/paper.js#1119]
Sorry to bring up another jspm-related issue...

The npm version of paper.js now works great! However, something seems to be wrong when creating a transpiled bundle.

The short: take a look at https://github.com/mrbannon/paper_jspm_transpile

The long: I'm using gulp-jspm to transpile my project (which includes paper.js 0.10.2) on the server side. With 0.9.25 there was never a problem. However, with 0.10.2 (after the great fix/conversation about jspm packaging...thanks again!) transpiled bundles that include paper.js 0.10.2 fail. The browser console (I'm using Chrome) comes back with:

Uncaught Module XXX not present

where XXX is some number. My naive guess is that a dependency is being referenced that is not included, maybe?

I've created an example that recreates this issue at https://github.com/mrbannon/paper_jspm_transpile. All the instructions are in README.md.

If you feel I should bring this up with babel or jspm maintainers, please let me know. (I thought I should ask you first as the version of paper.js is the variable that changed.)  Thanks.

**Comment by lehni** [comment]
My guess is that this is now happening because of the AMD "hack" described here, which is how paper is now loaded in JSPM: https://github.com/jspm/jspm-cli/issues/1981#issuecomment-234305752

There was the other approach with the override: https://github.com/jspm/jspm-cli/issues/1981#issuecomment-234297532

Can you try out this override file locally?

- Store this content in `override.json`

  ```json
  {
    "browser": {
      "canvas": false,
      "jsdom": false,
      "jsdom/lib/jsdom/living/generated/utils": false,
      "source-map-support": false,
      "./dist/node/window.js": false,
      "./dist/node/extend.js": false
    }
  }```

- Run this command:
   ```sh
   jspm install npm:paper -o override.json 
   ```

And let me know if that works? If it does, we'll have to get the jspm team to switch to this approach instead.

**Comment by lehni** [comment]
Update: Please note that I had to update the above override.json content. If you received an email notifcation, please make sure to copy the content from the website, not the email.

**Comment by mrbannon** [comment]
I'll try what you suggested on Monday. I should have mentioned that it also
happened when pulling from github instead of npm, so im not sure it's the
override issue.

**Comment by mrbannon** [comment]
Unfortunately it didn't work. I got the following error in console:

```
system.js:4 Uncaught (in promise) Error: (SystemJS) Node child_process module not supported in browsers.
	Evaluating http://xxx.xxx.xxx.xxx:9001/libs/github/jspm/nodelibs-child_process@0.1.0/index.js
	Evaluating http://xxx.xxx.xxx.xxx:9001/libs/github/jspm/nodelibs-child_process@0.1.0.js
	Evaluating http://xxx.xxx.xxx.xxx:9001/libs/npm/jsdom@9.4.1/lib/jsdom/living/xmlhttprequest.js
	Evaluating http://xxx.xxx.xxx.xxx:9001/libs/npm/jsdom@9.4.1/lib/jsdom/browser/Window.js
	Evaluating http://xxx.xxx.xxx.xxx:9001/libs/npm/jsdom@9.4.1/lib/jsdom.js
	Evaluating http://xxx.xxx.xxx.xxx:9001/libs/npm/jsdom@9.4.1.js
	Evaluating http://xxx.xxx.xxx.xxx:9001/libs/npm/paper@0.10.2/dist/node/window.js
	Error loading http://xxx.xxx.xxx.xxx:9001/Application.jst @ system.src.js:5123p @ system.src.js:5123(anonymous function) @ system.src.js:5123g @ system.src.js:5123b @ system.src.js:5123f @ system.src.js:5123m @ system.src.js:5123(anonymous function) @ system.src.js:5123
```

I made sure to use the override.json content from the site, not the email.

**Comment by mrbannon** [comment]
CLARIFICATION: The error I posted above happened when running `./node_modules/.bin/gulp`, which means server-side transpiling was NOT happening. When using `./node_modules/.bin/gulp dist:server` (server-side transpile), I get the following:

```
test.min.js:3826 Uncaught SyntaxError: missing ) after argument list
```

(transpiler error, it looks like in that case with the override you suggested)

**Comment by lehni** [comment]
Argh… It looks like it's still loading the jsdom module. I don't know what to suggest anymore. Could you follow up with the JSPM folks directly? I'm not using the manager myself, and this looks like it's clearly a problem on their side. I don't really have the time at the moment to dig further here, unfortunately.

Perhaps also test the upcoming JSPM 0.17 (currently in beta, I believe). If that works, maybe could switch to it early?

**Comment by mrbannon** [comment]
Will do. Thanks for the help.

Just so I'm clear, the issues is that jspm is trying to load the jsdom module which isn't there, correct?

**Comment by lehni** [comment]
Yes, I believe so… But it's hard for me to tell what's really going on. You could probably follow up on this issue here: https://github.com/jspm/jspm-cli/issues/1981

**Comment by lehni** [comment]
@mrbannon did you get anywhere here?

**Comment by mrbannon** [comment]
No, I haven't had time. Sorry :(.
